Fix "sdl.vapi:509.3-509.7: warning: argument `cheader' never used"
diff -Nur a/vapi/sdl.vapi b/vapi/sdl.vapi
--- a/vapi/sdl.vapi	2016-09-11 18:54:22.000000000 +0200
+++ b/vapi/sdl.vapi	2016-12-05 16:30:09.000000000 +0100
@@ -506,7 +506,7 @@
 		public weak SysWMmsg msg;
 	}// WMEvent
 
-	[CCode (cname="SDL_SysWMmsg", cheader="SDL_syswm.h")]
+	[CCode (cname="SDL_SysWMmsg", cheader_filename="SDL_syswm.h")]
 	public class SysWMmsg {
 	}// SysWMmsg
 
